bugsnag-android-core/src/main/java/com/bugsnag/android/ImmutableConfig.kt

```diff
@@ -83,9 +84,15 @@ internal data class ImmutableConfig(
 }
 
 internal fun convertToImmutableConfig(config: Configuration): ImmutableConfig {
+    val errorTypes = when {
+        config.autoDetectErrors -> config.enabledErrorTypes.copy()
+        else -> ErrorTypes(anrs = false, ndkCrashes = false, unhandledExceptions = false)
```

Comment:
With this implementation we'll need to remember to add any new errors types to this function (which is not located nearby) to allow the flag to work correctly.

An explicit call to `autoDetectErrors` in the places that it's used seems preferable to me as it makes the intent clearer. Alternatively, could `ErrorTypes` take the `autoDetectErrors` Boolean parameter that could be logically and'd with the defaults to turn them all off?

Reply:
Updated `ErrorTypes` to contain a non-public secondary constructor which switches off detection for all flags.